src/App.tsx

```diff
@@ -1,61 +1,149 @@
+import { FC, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+enum Field {
+  Title = 'title',
+  Select = 'select',
+}
+
+const findUser = (id: number) => (
+  usersFromServer.find(user => user.id === id) || null
+);
+
+const todos = (): Todo[] => todosFromServer.map(todo => ({
+  ...todo,
+  user: findUser(todo.userId),
+}));
+
+export const App: FC = () => {
+  const [vissibleTodos, setVissibleTodos] = useState<Todo[]>(todos);
+  const [inputTitle, setInputTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState<number>(0);
+  const [showTitleError, setShowTitleError] = useState(false);
+  const [showUserError, setShowUserError] = useState(false);
+
+  const newId = Math.max(...vissibleTodos.map(todo => todo.id)) + 1;
+
+  const reset = () => {
+    setInputTitle('');
+    setSelectedUser(0);
+  };
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+  ) => {
+    const { name, value } = event.target;
+
+    switch (name) {
+      case Field.Title:
+        setInputTitle(value.replace(/[^A-Za-zА-Яа-яЁё\s]/g, ''));
+        setShowTitleError(false);
+        break;
+
+      case Field.Select:
+        setSelectedUser(Number(event.target.value));
+        setShowUserError(false);
+        break;
+
+      // eslint-disable-next-line no-useless-escape
+      default: throw new Error('Something went wrong ¯\_(ツ)_/¯');
+        break;
+    }
+  };
+
+  const submitHandler = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const foundUser = findUser(Number(selectedUser));
+    const newTodo: Todo = {
+      id: newId,
+      title: inputTitle,
+      completed: false,
+      userId: selectedUser,
+      user: foundUser || null,
```

Comment:
you already return `null` if needed
```suggestion
      user: foundUser,
```

Reply:
Done